### PR TITLE
Move LaterPay enabled post type setting to pricing tab

### LINKED ISSUE #1016: Move “LaterPay Enabled Post Types” from LaterPay Settings page to the Pricing tab
### User Story
As a LaterPay Merchant, I do not want to have to hunt for hidden functionality. Especially not items essential to the initial plugin configuration.

### Acceptance Criteria

- Move “LaterPay Enabled Post Types” from LaterPay Settings page to the top of the Pricing tab
- No change in functionality

### Mock-up
image

### LINKED ISSUE #1018: Update Time Passes Description
### User Story
As a LaterPay merchant, the descriptions for time passes versus subscriptions is a bit confusing. I would like additional guidance on when to use each item so that I am not confused during my initial site set up.

### Acceptance Criteria
Update the Time Pass Description here:
image

**From**
> **Sell bundles of content**
> With Time Passes you can sell time-limited access to a category or your entire site. Time Passes do not renew automatically.
> 
> Click the "Create" button to add a Time Pass.

**To** 
> **Sell time-limited access to content**
> With Time Passes you can sell time-limited access to a category or your entire site. Time Passes do not renew automatically and are a great option for readers who are not quite ready to commit to a subscription.
>
> Click the “Create” button to add a Time Pass.

### PR DESCRIPTION
**Pull request by thrijith**
Fixes #1016 and Fixes #1018

- Move LaterPay enabled post type settings to the pricing tab.
- Remove related code from the Settings page.

### CODE REVIEW

File: laterpay/views/backend/pricing.php [hunk 1]
```diff
@@ -31,11 +31,64 @@ class="lp_plugin-mode-indicator"
...
+                        <td>
+                            <ul class="post_types">
+                                <?php
+                                foreach ( $all_post_types as $slug => $post_type_data ) {
+                                    if ( in_array( $slug, $hidden_post_types, true ) ) {
+                                        continue;
+                                    }
+                                    echo '<li><label title="' . esc_attr( $post_type_data->labels->name ) . '">';
+                                    echo '<input type="checkbox" name="laterpay_enabled_post_types[]" value="' . esc_attr( $slug ) . '" ';
+                                    if ( is_array( $enabled_post_types ) && in_array( $slug, $enabled_post_types, true ) ) {
```
Comment (PatelUtkarsh) [inline laterpay/views/backend/pricing.php]: use `checked()`?

File: laterpay/application/Helper/Option.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     * @param string $option_name  Option Name.
+     * @param mixed  $option_value Option Value.
+     *
+     * @return bool
+     */
+    public static function update_laterpay_option( $option_name, $option_value ) {
+        return update_option( $option_name, $option_value );
+    }
+
+}
```
Comment (PatelUtkarsh) [inline laterpay/application/Helper/Option.php]: Should add new line.

File: laterpay/views/backend/pricing.php [hunk 3]
```diff
@@ -31,11 +31,64 @@ class="lp_plugin-mode-indicator"
...
+                    esc_html__( 'All WordPress Posts, Pages and Media will display the pricing options below unless this is overridden by setting an alternate', 'laterpay' ),
+                    esc_html__( 'Individual Article Price on the specific page or by a rule established on the Custom Pricing Tab.', 'laterpay' )
+                );
+                ?>
+            </p>
+
+            <form id="lp_js_globalEnabledPostTypesForm" method="post" action="">
+                <input type="hidden" name="form"    value="update_enabled_post_types">
+                <input type="hidden" name="action"  value="laterpay_enabled_post_types">
+                <?php if ( function_exists( 'wp_nonce_field' ) ) { wp_nonce_field( 'laterpay_form' ); } ?>
```
Comment (PatelUtkarsh) [inline laterpay/views/backend/pricing.php]: Any reason for function exists check?

File: laterpay/application/Form/EnabledPostType.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public function init() {
+
+        $this->set_field(
+            'form',
+            array(
+                'validators' => array(
+                    'is_string',
+                    'cmp' => array(
+                        array(
+                            'like' => 'update_enabled_post_types',
```
Comment (PatelUtkarsh) [inline laterpay/application/Form/EnabledPostType.php]: Does it compare it with existing post type? 
If not can we compare with all current post type just to be safe? `get_post_types( array( ), 'objects' );`

File: laterpay/views/backend/pricing.php (hunk 3 above)
Comment (thrijith) [reply laterpay/views/backend/pricing.php]: I checked existing forms for nonce implementation. The check is added almost everywhere. Can remove it, no issue.

File: laterpay/views/backend/pricing.php [hunk 5]
```diff
@@ -66,14 +66,14 @@ class="lp_plugin-mode-indicator"
...
-                                    if ( is_array( $enabled_post_types ) && in_array( $slug, $enabled_post_types, true ) ) {
-                                        echo 'checked';
-                                    }
-                                    echo '>';
-                                    echo '<span>' . esc_html( $post_type_data->labels->name ) . '</span>';
-                                    echo '</label></li>';
+                                    ?>
+                                    <li>
+                                        <label title="<?php echo esc_attr( $post_type_data->labels->name ); ?>">
+                                            <input type="checkbox" name="laterpay_enabled_post_types[]" value="<?php echo esc_attr( $slug ); ?>" <?php checked( is_array( $enabled_post_types ) ? in_array( $slug, $enabled_post_types, true ) : '' ); ?>>
```
Comment (PatelUtkarsh) [inline laterpay/views/backend/pricing.php]: Should it be ? `checked( ( is_array( $enabled_post_types ) && in_array( $slug, $enabled_post_types, true ) );`

Alternatively you can put condition in variable and use it. @thrijith

**Pull request merged**